### ISSUE TITLE
I cant install openmm 8.1.1 and CUDA 12.0.*

### ISSUE THREAD

**Issue by RaulPPelaez** [conda-forge/openmm-feedstock#127]
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

This command should correctly install openmm 8.1.1 along cuda 12.0
```shell
$ mamba create -n testop openmm==8.1.1 cuda-version==12.0.*
WARNING: A conda environment already exists at '/home/raul/miniforge3/envs/testop'
Remove existing environment (y/[n])? y


Looking for: ['openmm==8.1.1', 'cuda-version=12.0']

conda-forge/noarch                                  13.3MB @  20.8MB/s  0.7s
conda-forge/linux-64                                32.1MB @  34.5MB/s  1.0s
Could not solve for environment specs
The following packages are incompatible
├─ cuda-version 12.0**  is installable and it requires
│  └─ cudatoolkit 12.0|12.0.* , which can be installed;
└─ openmm 8.1.1  is not installable because there are no viable options
   ├─ openmm 8.1.1 would require
   │  └─ libcufft >=11.0.8.103,<12.0a0  but there are no viable options
   │     ├─ libcufft 11.0.12.1 would require
   │     │  └─ cuda-version >=12.3,<12.4.0a0 , which conflicts with any installable versions previously reported;
   │     └─ libcufft 11.0.8.103 would require
   │        └─ cuda-version >=12.2,<12.3.0a0 , which conflicts with any installable versions previously reported;
   ├─ openmm 8.1.1 would require
   │  └─ cudatoolkit >=11.2,<12 , which conflicts with any installable versions previously reported;
   └─ openmm 8.1.1 would require
      └─ cudatoolkit >=11.8,<12 , which conflicts with any installable versions previously reported.
```
Conda also fails, just to discard that :P 
I am reopening https://github.com/conda-forge/openmm-torch-feedstock/issues/48 here, since I just realized I opened it in openmm-torch by mistake.

### Installed packages

```shell
N/A
```

### Environment info

```shell
N/A
```

**Comment by jakirkham** [comment]
Thanks Raul! 🙏

Think what may be happening is the build environment is using CUDA 12.0 whereas the host environment is using CUDA 12.x where x is latest. As a result the pinnings on CUDA 12.x libraries are determined by the host environment, which are too new

To fix this, think we need to constrain the CUDA version used in `requirements/host` to match the CUDA version used in `requirements/build` as show below. Then `requirements/run` will then have lower version constraints on these libraries, which will make them easier to install with a wider range of CUDA versions 

```yaml
    - cuda-version {{ cuda_compiler_version }}  # [(cuda_compiler_version or "None") != "None"]
```

Added a note to the OP in issue ( https://github.com/conda-forge/conda-forge.github.io/issues/1963 ) somewhat recently entitled "On CUDA 12.x", which explains this a bit more

Hope that helps!

**Comment by RaulPPelaez** [comment]
Ok so I should have the cura-version line in both build and host but not run, right? 
I wanna make sure since we do these cuda-version shenanigans in several feedstocks...

**Comment by jakirkham** [comment]
`cuda-version` should only be needed in `requirements/host`

With `requirements/build`, `{{ compiler("cuda") }}` pulls `cuda-version` in for us on CUDA 12

CUDA 11 shouldn't have this issue as everything is bundled in `cudatoolkit`

**Comment by RaulPPelaez** [comment]
This has been fixed by #128 . Thanks guys!

**Comment by jakirkham** [comment]
Great, thanks for confirming Raul! 🙏
